src/build_vehicle_gui.cpp

```diff
@@ -854,28 +858,41 @@ static int DrawAircraftPurchaseInfo(int left, int right, int y, EngineID engine_
 	return y;
 }
 
+
 /**
- * Display additional text from NewGRF in the purchase information window
- * @param left   Left border of text bounding box
- * @param right  Right border of text bounding box
- * @param y      Top border of text bounding box
- * @param engine Engine to query the additional purchase information for
- * @return       Bottom border of text bounding box
+ *
  */
-static uint ShowAdditionalText(int left, int right, int y, EngineID engine)
+static std::tuple<bool, std::string> GetNewGRFAdditionalText(EngineID engine)
```

Comment:
Clippy-mode: "it looks like you're trying to implement `std::optional`" ;)
But seriously, I think `std:optional` is what you conceptually want here. Also, is the comment of this function missing?

Sadly we're still at C++17 and not C++23, as then we could use the monadic functionality and write `ShowAdditionalText` as:
```
return GetNewGRFAdditionalText(engine)
        .and_then([=] (auto &text) { return DrawStringMultiLine(left, right, y, INT32_MAX, text, TC_BLACK); }
        .value_or(y);
```

Reply:
Thanks for the advice. I've learned a lot from this review. 🙂 I've never used `std::optional` but it's exactly what I want.